internal/web/results.go

```diff
@@ -185,6 +211,50 @@ func renderInProgress(w http.ResponseWriter, r *http.Request, badge []byte, vali
 	}
 }
 
+func myReadDir(dirname string) ([]os.FileInfo, error) {
+	f, err := os.Open(dirname)
+	if err != nil {
+		return nil, err
+	}
+	list, err := f.Readdir(-1)
+	f.Close()
```

Comment:
Move the line one up and use `defer f.Close()` to ensure `Close` is called even if horrible things happen to `Readdir`.